cmd/cem.go

```diff
@@ -0,0 +1,395 @@
+		}
+
+		cemQuery := &cem.EntityQuery{
+			Platform:  CemPlatform,
+			AccountID: CemAccountID,
+			EntityID:  CemEntityID,
+		}
+
+		cemResult, err := cem.GetEntityDetail(token, cemQuery)
+		if err != nil {
+			log.Fatalln(err)
+			return
+		}
+
+		prettyprint.PrintJSON(cemResult)
+	},
+}
+
+var cemGetEntitiesCmd = &cobra.Command{
+	Use:   "get-entities",
+	Short: "Get Entities",
+	Long: `Search for entities on any platform and retrieve entity details.
+	
+	Example Usage:
+	$ cybr cem get-entities -p PLATFORM -a ACCOUNT_ID`,
+	Run: func(cmd *cobra.Command, args []string) {
+		token, err := GetToken()
+		if err != nil {
+			log.Fatalf("Failed to retrieve token file at %s. %s\n", CemSessionTokenPath, err)
+		}
+
+		if CemNonFullAdmin && CemFullAdmin {
+			log.Fatal("Cannot set both --non-full-admin and --full-admin at the same time")
+		}
+
+		if CemNonShadowAdmin && CemShadowAdmin {
+			log.Fatal("Cannot set both --non-shadow-admin and --shadow-admin at the same time")
+		}
+
+		inputFullAdmin := ""
+		inputShadowAdmin := ""
+
+		if CemFullAdmin {
+			inputFullAdmin = "true"
+		}
+		if CemNonFullAdmin {
+			inputFullAdmin = "false"
+		}
+		if CemShadowAdmin {
+			inputShadowAdmin = "true"
+		}
+		if CemNonShadowAdmin {
+			inputShadowAdmin = "false"
+		}
+
+		cemQuery := &cem.GetEntitiesQuery{
+			Platform:    CemPlatform,
+			AccountId:   CemAccountID,
+			FullAdmin:   inputFullAdmin,
+			ShadowAdmin: inputShadowAdmin,
+			NextToken:   CemNextToken,
+		}
+
+		cemResult, err := cem.GetEntities(token, cemQuery)
+		if err != nil {
+			log.Fatalln(err)
+			return
+		}
+
+		prettyprint.PrintJSON(cemResult)
+	},
+}
+
+func init() {
+
+	// Login
+	cemLoginCmd.Flags().StringVarP(&CemOrganization, "organization", "o", "", "Username to logon PAS REST API using")
+	cemLoginCmd.MarkFlagRequired("organization")
+	cemLoginCmd.Flags().BoolVar(&CemNonInteractive, "non-interactive", false, "If detected, will retrieve the API key from the CEM_APIKEY environment variable")
+
+	// Get Entities
+	cemGetEntitiesCmd.Flags().StringVarP(&CemPlatform, "platform", "p", "", "Platform Name")
+	cemGetEntitiesCmd.Flags().StringVarP(&CemAccountID, "account-id", "a", "", "Account ID")
+	cemGetEntitiesCmd.Flags().BoolVar(&CemNonFullAdmin, "non-full-admin", false, "Get non-full admin entities only. Cannot be used with --full-admin")
+	cemGetEntitiesCmd.Flags().BoolVar(&CemNonShadowAdmin, "non-shadow-admin", false, "Get non-shadow admin entities only.  Cannot be used with --shadow-admin")
+	cemGetEntitiesCmd.Flags().BoolVar(&CemFullAdmin, "full-admin", false, "Get full admin entities only.  Cannot be used with --non-full-admin")
+	cemGetEntitiesCmd.Flags().BoolVar(&CemShadowAdmin, "shadow-admin", false, "Get shadow admin entities only.  Cannot be used with --non-shadow-admin")
+	cemGetEntitiesCmd.Flags().StringVarP(&CemNextToken, "next-token", "n", "", "The token for paging the entities.")
+
+	// Get Entity Detail
```

Comment:
Are all 3 values required for `platform`, `account-id`, and `entity-id`?  I feel like maybe only `account-id` would need to be required to get Entity Detail.  I don't understand why all 3 would be required.  I guess maybe this was the method preferred over a unique account key of some kind?

Reply:
You would think that, however that is not the case :D

Reply:
![this-is-cybi-125](https://user-images.githubusercontent.com/1924063/114595075-f9b0c280-9c5b-11eb-932e-d2d50ea23214.gif)